db/substate_db.go

```diff
@@ -24,6 +25,8 @@ type SubstateDB interface {
 	// GetSubstate gets the Substate for given block and tx number.
 	GetSubstate(block uint64, tx int) (*substate.Substate, error)
 
+	GetBlockSubstates(block uint64) map[int]*substate.Substate
```

Comment:
nit: Could you add documentation what does this function do please?

Reply:
Done